File: 0179-largest-number/0179-largest-number.ts
function largestNumber(nums: number[]): string {
    let str_list = nums.map(num => "" + num);
    str_list.sort((a,b)=> {
        if (a + b > b + a) return -1
        else return 1
    });
    return str_list.join("").charAt(0) == '0'? "0":str_list.join("")
};

File: 0179-largest-number/0179-largest-number.py
class Solution:
    def largestNumber(self, nums: List[int]) -> str:
        str_array = list(map(str, nums))
        str_array.sort(key = cmp_to_key(lambda a,b :(b+a>a+b) - (a+b> b+a)))
        res = ('').join(str_array)
        return '0' if res[0] == '0'else res
        